qa/L0_input_validation/input_validation_test.py

```diff
@@ -116,101 +118,113 @@ def test_input_validation_all_optional(self):
+                triton_client.infer(model_name=model_name, inputs=inputs)
 
-        # Input size is less than expected
-        inputs = get_input_array(input_size - 2, np_dtype_string)
-        # Compromised input shape
-        inputs[0].set_shape((1, input_size))
-        inputs[1].set_shape((1, input_size))
-        with self.assertRaises(InferenceServerException) as e:
-            triton_client.infer(model_name=model_name, inputs=inputs)
-        err_str = str(e.exception)
-        self.assertIn(
-            f"expected {input_size} string elements for inference input 'INPUT1', got {input_size-2}",
-            err_str,
-        )
+                # Compromised input shape
+                inputs[0].set_shape([1, 8])
 
-        # Input size is greater than expected
-        inputs = get_input_array(input_size + 2, np_dtype_string)
-        # Compromised input shape
-        inputs[0].set_shape((1, input_size))
-        inputs[1].set_shape((1, input_size))
-        with self.assertRaises(InferenceServerException) as e:
-            triton_client.infer(model_name=model_name, inputs=inputs)
-        err_str = str(e.exception)
-        self.assertIn(
-            f"expected {input_size} string elements for inference input 'INPUT1', got {input_size+2}",
-            err_str,
-        )
+                with self.assertRaises(InferenceServerException) as e:
+                    triton_client.infer(model_name=model_name, inputs=inputs)
+                err_str = str(e.exception)
+                self.assertIn(
+                    f"input 'INPUT0' got unexpected elements count 16, expected 8",
+                    err_str,
+                )
+
+            if client_type == "http":
+                triton_client = tritonhttpclient.InferenceServerClient("localhost:8000")
+            else:
+                triton_client = tritongrpcclient.InferenceServerClient("localhost:8001")
+
+            # Example using BYTES input tensor with utf-8 encoded string that
+            # has an embedded null character.
+            null_chars_array = np.array(
+                ["he\x00llo".encode("utf-8") for i in range(16)], dtype=np.object_
+            )
+            null_char_data = null_chars_array.reshape([1, 16])
+            identity_inference(triton_client, null_char_data, True)  # Using binary data
+            identity_inference(triton_client, null_char_data, False)  # Using JSON data
+
+            # Example using BYTES input tensor with 16 elements, where each
+            # element is a 4-byte binary blob with value 0x00010203. Can use
+            # dtype=np.bytes_ in this case.
+            bytes_data = [b"\x00\x01\x02\x03" for i in range(16)]
+            np_bytes_data = np.array(bytes_data, dtype=np.bytes_)
+            np_bytes_data = np_bytes_data.reshape([1, 16])
+            identity_inference(triton_client, np_bytes_data, True)  # Using binary data
+            identity_inference(triton_client, np_bytes_data, False)  # Using JSON data
```

Comment:
What is this testing?

Reply:
Copied from `client/src/python/examples/simple_http_string_infer_client.py`. It looks like the example demonstrated two ways of preparing string input data. I'll remove one of them.